### PR TITLE
Added ConnectionType functionality

### PR DESCRIPTION
**Pull request by Whirvis**
This functionality allows for implementations that implement this feature to get the type of implementation of another connection (if that connection's API implements it). Implementing this also allows other API's that implement this feature to see what connection type (implementation) you are using. The main usage of this is for statistical reasons, as well as possibly even implementing enhancements if both and the client and server are implemented. If there are any issues you have, please let me know and I will be happy to fix them.

### CODE REVIEW

**Comment by Aericio** [comment]
implementation^100

**Comment by Whirvis** [comment]
I have noticed an issue with this: At the moment, the UUID is written as a string rather than two longs by the most significant bits and then the least significant bits. I will get this fixed as soon as possible.

**Comment by dktapps** [comment]
This code is full of syntax errors.

File: src/protocol/ConnectionType.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+#include <rules/RakLibPacket.h>
+
+use raklib\RakLib;
+
+class ConnectionType{
+	
+	public const MAX_METADATA_VALUES = 0xff;
+	
+	public static function createMetaData(string... $metadata) : array{
```
Comment (SOF3) [inline src/protocol/ConnectionType.php]: Inconsistent code style with other parts of @pmmp

**Comment by Whirvis** [comment]
@dktapps Thank you for the response. I made this in a plain text editor, it was really hard for me to know if something had a syntax error or not. What IDE do you guys use when coding PMMP?

File: src/protocol/ConnectionType.php (hunk 1 above)
Comment (Whirvis) [reply src/protocol/ConnectionType.php]: I'm not sure how this is inconsistent with other parts of PMMP, could you elaborate please?

**Comment by Aericio** [comment]
They use PHPStorm by JetBrains (or at least SOF3 does).
https://www.jetbrains.com/phpstorm/

It's free if you sign up for an education account (trial is 30 days, though).

File: src/protocol/ConnectionType.php (hunk 1 above)
Comment (SOF3) [reply src/protocol/ConnectionType.php]: In PHP, the `...` is defined to be on the left of the parameter name, not to be on the right of the data type. We usually put the space between the type and the `...`.

Comment (Whirvis) [reply src/protocol/ConnectionType.php]: Alright, thanks for the info!

**Comment by Whirvis** [comment]
Alright, I believe that I have fixed the main issues when it comes to syntax, code style inconsistency, and overall functionality. I know the "..." after the string argument is still not to the left, however JetBrains gave me an error when I tried to do so. I did put a space in between the string type indicator and the ellipses, though.

Please let me know if there's anything else you need me to do to get it working. I apologize for taking so long to update this and get back to. Thank you for getting back to me the first time, I really appreciate it!

File: src/protocol/ConnectionType.php (hunk 1 above)
Comment (Whirvis) [reply src/protocol/ConnectionType.php]: I added the space like you suggested! :)

**Comment by Whirvis** [comment]
Deleted repository containing code for fork

**Pull request closed without merging**